### PR TITLE
Added support credential env variables modules/roles

### LINKED ISSUE #554: [BUG]: dellemc.openmanage 8.2.0 ignores ```NO_PROXY```
### Bug Description

In our environment we need to use a proxy to access public internet resources. Our Dell OpenManage Enterprise installation however can be reached directly and not via the proxy server. Therefore on our machines we have the ```http_proxy```, ```HTTP_PROXY```, ```https_proxy```, ```HTTPS_PROXY```, ```no_proxy``` and ```NO_PROXY``` environment variables defined. ```http_proxy```, ```HTTP_PROXY```, ```https_proxy``` and ```HTTPS_PROXY``` point to our proxy server while ```no_proxy``` and ```NO_PROXY``` contains proxy bypass exceptions for internal resources in the format ```.company.com```

With ```dellemc.openmanage``` modules 8.0.0 and 8.1.0 this worked fine and our ansible host was able to communicate with our Dell OpenManage Enterprise installation. Unfortunately with ```dellemc.openmanage``` 8.2.0 we receive the following error message:

`<urlopen error _ssl.c:975: The handshake operation timed out>`

I did some troubleshooting and for me it looks like the ```ome.py``` in version 8.2.0 does a DNS lookup of the URL first and then connects to the IP address of the Dell OpenManage installation instead of the DNS FQDN. Therefore our proxy bypass exceptions do not match anymore and the ```dellemc.openmanage``` tries to contact our OME through our proxy which fails.

As a workaround I added the IP address of our OME installation to the NO_PROXY environment variable as well and afterwards ```dellemc.openmanage``` is able to contact OME again.  

### Component or Module Name

dellemc.openmanage.ome_device_info

### Ansible Version

Ansible core 2.15.4

### Python Version

Pyhton 3.11.2

### iDRAC/OME/OME-M version

OME 3.10.2
dellemc.openmanage 8.2.0

### Operating System

Debian 11

### Playbook Used

```yaml
- name: Retrieve basic managed device information from Dell Openmanage Enterprise
  dellemc.openmanage.ome_device_info:
    hostname: "{{ ome_server }}"
    username: "{{ domain_user ~ '@' ~ domain }}"
    password: "{{ domain_password }}"
    ca_path: "{{ ome_ca_path }}"
    validate_certs: "{{ ome_validate_certs }}"
    system_query_options:
      filter: "DeviceName eq '{{ inventory_hostname }}'"
  delegate_to: localhost
  register: update_firmware_ome_device_info
```

### Logs

```
TASK [update_firmware : Retrieve basic managed device information from Dell Openmanage Enterprise] ***************************************************************************************************************************************
task path: /scripts/ansible/roles/update_firmware/tasks/main.yml:8
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: root
<localhost> EXEC /bin/sh -c 'echo ~root && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /root/.ansible/tmp `"&& mkdir "` echo /root/.ansible/tmp/ansible-tmp-1694755665.3208275-79-183326375421279 `" && echo ansible-tmp-1694755665.3208275-79-183326375421279="` echo /root/.ansible/tmp/ansible-tmp-1694755665.3208275-79-183326375421279 `" ) && sleep 0'
Using module file /opt/ansible_venv/lib/python3.11/site-packages/ansible_collections/dellemc/openmanage/plugins/modules/ome_device_info.py
<localhost> PUT /root/.ansible/tmp/ansible-local-6101xabs_a/tmpu4iof2tu TO /root/.ansible/tmp/ansible-tmp-1694755665.3208275-79-183326375421279/AnsiballZ_ome_device_info.py
<localhost> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1694755665.3208275-79-183326375421279/ /root/.ansible/tmp/ansible-tmp-1694755665.3208275-79-183326375421279/AnsiballZ_ome_device_info.py && sleep 0'
<localhost> EXEC /bin/sh -c '/opt/ansible_venv/bin/python3 /root/.ansible/tmp/ansible-tmp-1694755665.3208275-79-183326375421279/AnsiballZ_ome_device_info.py && sleep 0'
<localhost> EXEC /bin/sh -c 'rm -f -r /root/.ansible/tmp/ansible-tmp-1694755665.3208275-79-183326375421279/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
  File "/tmp/ansible_dellemc.openmanage.ome_device_info_payload_ktgztv2v/ansible_dellemc.openmanage.ome_device_info_payload.zip/ansible_collections/dellemc/openmanage/plugins/modules/ome_device_info.py", line 389, in main
  File "/tmp/ansible_dellemc.openmanage.ome_device_info_payload_ktgztv2v/ansible_dellemc.openmanage.ome_device_info_payload.zip/ansible_collections/dellemc/openmanage/plugins/module_utils/ome.py", line 206, in __enter__
    resp = self.invoke_request('POST', path, data=payload)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/ansible_dellemc.openmanage.ome_device_info_payload_ktgztv2v/ansible_dellemc.openmanage.ome_device_info_payload.zip/ansible_collections/dellemc/openmanage/plugins/module_utils/ome.py", line 196, in invoke_request
    raise err
  File "/tmp/ansible_dellemc.openmanage.ome_device_info_payload_ktgztv2v/ansible_dellemc.openmanage.ome_device_info_payload.zip/ansible_collections/dellemc/openmanage/plugins/module_utils/ome.py", line 193, in invoke_request
    resp = open_url(url, data=data, **url_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/ansible_dellemc.openmanage.ome_device_info_payload_ktgztv2v/ansible_dellemc.openmanage.ome_device_info_payload.zip/ansible/module_utils/urls.py", line 1669, in open_url
    return Request().open(method, url, data=data, headers=headers, use_proxy=use_proxy,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/ansible_dellemc.openmanage.ome_device_info_payload_ktgztv2v/ansible_dellemc.openmanage.ome_device_info_payload.zip/ansible/module_utils/urls.py", line 1561, in open
    r = urllib_request.urlopen(request, None, timeout)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/urllib/request.py", line 216, in urlopen
    return opener.open(url, data, timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/urllib/request.py", line 519, in open
    response = self._open(req, data)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/urllib/request.py", line 536, in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/urllib/request.py", line 496, in _call_chain
    result = func(*args)
             ^^^^^^^^^^^
  File "/tmp/ansible_dellemc.openmanage.ome_device_info_payload_ktgztv2v/ansible_dellemc.openmanage.ome_device_info_payload.zip/ansible/module_utils/urls.py", line 605, in https_open
    return self.do_open(self._build_https_connection, req)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/urllib/request.py", line 1351, in do_open
    raise URLError(err)
fatal: [somehost.company.com -> localhost]: FAILED! => changed=false
  failed_when_result: true
  invocation:
    module_args:
      ca_path: /usr/local/share/ca-certificates/OUR-CA.crt
      fact_subset: basic_inventory
      hostname: ome.company.com
      password: VALUE_SPECIFIED_IN_NO_LOG_PARAMETER
      port: 443
      system_query_options:
        device_id: null
        device_service_tag: null
        filter: DeviceName eq 'somehost.company.com'
        inventory_type: null
      timeout: 30
      username: admin@company.com
      validate_certs: false
  msg: '<urlopen error _ssl.c:975: The handshake operation timed out>'
```

### Steps to Reproduce

- Have a network environment which requires a proxy to access the internet
- Have a local Dell OME installation that needs to be accessed directly, not via proxy
- Have local domain names excluded from proxy by setting `no_proxy` or `NO_PROXY` environment variables, e.g. `.company.com`
- Have dellemc.openmanage 8.2.0 installed
- Run above task
- Observe connections from ansible host to proxy trying to access IP address from OME in tcpdump

### Expected Behavior

Connections to the domain name of the OME installation will honor proxy bypass exceptions and be created directly (as in `dellemc.openmanage` `8.0.0` or `8.1.0`

### Actual Behavior

`dellemc.openmanage` `8.2.0` tries to access the OME system through proxy because IP addresses proxy bypass exceptions are not set in `no_proxy` or `NO_PROXY` environment variables

### Screenshots

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by singh-lovepreet1**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

### CODE REVIEW

**Comment by ansible-collections-svc** [comment]
Can one of the admins verify this patch?

File: roles/idrac_gather_facts/tasks/get_attributes_info.yml [hunk 1]
```diff
@@ -5,8 +5,8 @@
     validate_certs: "{{ validate_certs }}"
     ca_path: "{{ ca_path | default(omit) }}"
     method: "{{ idrac_gather_facts_uri_method }}"
-    user: "{{ username }}"
-    password: "{{ password }}"
+    user: "{{ username | default(lookup('env', 'iDRAC_USERNAME')) }}"
```
Comment (jagadeeshnv) [inline roles/idrac_gather_facts/tasks/get_attributes_info.yml]: "i" should be in UPPERCASE right?

**Comment by sachin-apa** [comment]
run e2e

File: plugins/module_utils/dellemc_idrac.py [hunk 2]
```diff
@@ -81,14 +82,8 @@ def __init__(self, module_params):
...
-        try:
-            data = socket.getaddrinfo(self.idrac_ip, self.idrac_port)
-            if "AF_INET6" == data[0][0]._name_:
-                ip_byte = socket.inet_pton(socket.AF_INET6, self.idrac_ip)
-                ip_addr = socket.inet_ntop(socket.AF_INET6, ip_byte)
-                self.idrac_ip = "{0}".format(ip_addr)
-        except socket.gaierror:
-            pass
+        self.idrac_ip = config_ipv6(self.idrac_ip, omsdk_used=True)
+        self.idrac_ip = self.idrac_ip.strip('[]')
```
Comment (jagadeeshnv) [inline plugins/module_utils/dellemc_idrac.py:84]: config_ipv6 just adds the "square brackets" and what is the need to strip it?

**Review by jagadeeshnv — CHANGES_REQUESTED** [review]

File: tests/unit/plugins/modules/test_ome_firmware_catalog.py [hunk 3]
```diff
@@ -863,12 +863,18 @@ def test_ome_catalog_firmware_argument_exception_case4(self, ome_default_args):
...
-    def test_ome_catalog_invalid_hostname(self, ome_default_args, param):
+    @pytest.mark.parametrize("param", [{"hostname": "invalid-host-abcd"}])
+    def test_ome_catalog_invalid_hostname_case1(self, ome_default_args, param):
         # To verify invalid IP or hostname in module_utils/ome
-        ome_default_args.update({"hostname": param['hostname'], "catalog_name": "catalog1", "repository_type": "HTTPS"})
+        ome_default_args.update({"hostname": param['hostname'], "catalog_name": "catalog1", "repository_type": "HTTPS", "ca_path": ""})
         result = self._run_module(ome_default_args)
         assert result["unreachable"] is True
-        assert "Unable to resolve hostname or IP" in result['msg']
+        assert "error" in result['msg']
```
Comment (Jennifer-John) [inline tests/unit/plugins/modules/test_ome_firmware_catalog.py:872]: revert this change? now we are asserting only for "error" in result

File: roles/idrac_gather_facts/tasks/get_attributes_info.yml (hunk 1 above)
Comment (singh-lovepreet1) [reply roles/idrac_gather_facts/tasks/get_attributes_info.yml]: This is the outdated version. It was resolved.

File: tests/unit/plugins/modules/test_ome_firmware_catalog.py (hunk 3 above)
Comment (singh-lovepreet1) [reply tests/unit/plugins/modules/test_ome_firmware_catalog.py:872]: Yes, earlier we were raising an exception with the custom message which included "Unable to resolve hostname or IP". Now, we have removed that code block. So, now we are getting some internal error.

File: plugins/module_utils/dellemc_idrac.py (hunk 2 above)
Comment (singh-lovepreet1) [reply plugins/module_utils/dellemc_idrac.py:84]: Removed unnecessary code snippets.

**Review by jagadeeshnv — DISMISSED** [review]

**Review by sachin-apa — DISMISSED** [review]

**Review by rajshekarp87 — DISMISSED** [review]

**Review by sachin-apa — APPROVED** [review]

**Review by ShivamSh3 — APPROVED** [review]

**Review by jagadeeshnv — APPROVED** [review]
